accepted/future-releases/unquoted-imports/feature-specification.md

```diff
@@ -514,33 +500,33 @@ recommended URI syntax and would not be affected by this change:
    4466 ( 25.239%): library name  ============
 ```
 
-In total, only about 1% of directives are `part of` with a library name:
-
-```
--- URI (443733 total) --
- 352744 ( 79.495%): import                     ===========================
-  55471 ( 12.501%): export                     =====
-  17823 (  4.017%): part                       ==
-  13229 (  2.981%): part of with uri           =
-   4466 (  1.006%): part of with library name  =
-```
+If we reinterpreted that existing syntax to refer to another package instead of
+a library within the current package, user breakage is likely and would be very
+confusing.
 
-Given that, I propose that we make a **breaking change** and remove support for
-the long-deprecated library name syntax from `part of` directives. An unquoted
-series of identifiers after `part of` then gets unambiguously interpreted as
-this proposal's semantics. In other words, `part of foo.bar;` is part of the
-library at `package:foo/bar.dart`, not part of the library with name `foo.bar`.
+To avoid potential user confusion and breakage, the new syntax in this proposal
+isn't allowed in `part` and `part of` directives. Further, we make a
+**language-versioned breaking change** and remove support for unquoted library
+identifiers in `part of` directives completely. That way, a user never sees
+syntax in a `part of` file that *looks* like a package path but means something
+else.
 
-Users affected by the breakage can and should update their `part of` directive
-to point to the URI of the library that the file is a part of, using either the
-quoted or unquoted syntax.
+Since part files should always be part of the same package that contains the
+library that owns them, the new unquoted syntax is of limited use for part files
+anyway. Quoted relative URIs are shorter and more idiomatic.
```

Comment:
(It's not actually specified anywhere, yet, that part files *must* be in the same library as the library they're part of. I added that rule to the enhanced parts spec. It'd be fine to add it as a normative rule here too, so whoever lands first gets the rule enforced. The sooner the better.)

Reply:
I meant "should" in the sense of "good practice" and not normatively in the language spec. I agree the language should enforce that, but given that it has almost no notion of a package anyway, I'm not sure how I'd specify that. Reworded to make it clear I wasn't being spec-normative.